sale_coupon_order_line_link/views/sale_order_views.xml

```diff
@@ -5,46 +5,44 @@
         <field name="inherit_id" ref="sale.view_order_form" />
         <field name="arch" type="xml">
             <xpath expr="//page[@name='other_information']" position="inside">
-                <group
-                    attrs="{'invisible': [('no_code_promo_program_ids', '=', []), ('code_promo_program_id', '=', False), ('applied_coupon_ids', '=', []), ('generated_coupon_ids', '=', [])]}"
-                >
+                <group attrs="{'invisible': [('applied_coupon_ids', '=', [])]}">
                     <group name="promo_info" string="Promotions">
-                        <field
+                        <!-- <field
```

Comment:
This view needs to be revised because the fields have changed and may need to be added to the sale.order model to display them.

Reply:
These fields no longer exist in the sale.order model nor have they been renamed, I would have to add them to the model to get the promotion/coupon applied and in the case that a coupon is generated also add a field to display those generated coupons. I don't know if this is necessary to complete the functionality of the module or if this would be a parallel functionality.